boot/assets_test.go

```diff
@@ -283,6 +296,86 @@ var (
 	}
 )
 
+func (s *assetsSuite) TestInstallObserverCurrentTrustedBootAssets(c *C) {
+	d := c.MkDir()
+	uc20Model := boottest.MakeMockUC20Model()
+
+	// pretend grub is used for all variants
+	c.Assert(ioutil.WriteFile(filepath.Join(d, "grub.conf"), nil, 0755), IsNil)
+
+	writeChange := &gadget.ContentChange{
+		// file that contains the data of the installed file
+		After: filepath.Join(d, "realasset"),
+		// there is no original file in place
+		Before: "",
+	}
+
+	data := []byte("realasset")
+	// SHA3-384
+	dataHash := "467b6f75ec7de3f95dfa23b78873a3b4284a0f6ce905749df788323c81371d60628a2d0839c81b02ee2fccd645e16474"
+	err := ioutil.WriteFile(filepath.Join(d, "realasset"), data, 0644)
+	c.Assert(err, IsNil)
+
+	for _, useEncryption := range []bool{true, false} {
+		installObserverOpts := &boot.InstallObserverOptions{
+			UseEncryption: useEncryption,
+		}
+		obs, err := boot.TrustedAssetsInstallObserverForModel(uc20Model, d, installObserverOpts)
+		c.Assert(err, IsNil)
+
+		// no assets since we haven't observed anything and didn't get any from the
+		// setup opts
+		c.Assert(obs.CurrentTrustedBootAssetsMap(), HasLen, 0)
+		c.Assert(obs.CurrentTrustedRecoveryBootAssetsMap(), HasLen, 0)
+
+		// create an observer with some pre-defined assets
+		installObserverOpts.TrackedAssets = boot.AssetsMap{
+			"some-asset": []string{"foo", "bar"},
```

Comment:
given that we're restoring those from the state, I think it's reasonable to try a test case where the assets map that we set already contains grubx64.efi with the right hash, so that observing its install effectively does nothing

Reply:
good idea, I will add that test too